### PR TITLE
Add `identifierutil` package

### PR DESCRIPTION
**Pull request by snelusha**
This pull request introduces a new package, `identifierutil`, that provides comprehensive utilities for encoding and decoding identifiers, with robust handling for special characters, Unicode codepoints, and JVM-reserved characters. It also adds an extensive suite of unit tests to ensure correctness and reliability of these utilities.

### Identifier encoding/decoding utilities

* Added `identifierutil/util.go` with functions for encoding and decoding identifiers, including handling of special characters, JVM-reserved characters, Unicode escapes, and generated method names. This includes functions like `EncodeFunctionIdentifier`, `EncodeNonFunctionIdentifier`, `DecodeIdentifier`, `UnescapeJava`, and related helpers.

### Test coverage

* Added `identifierutil/util_test.go` with thorough unit tests for all major utility functions, covering edge cases and various scenarios for escaping, unescaping, encoding, and decoding identifiers.

### CODE REVIEW

File: identifierutil/util.go [hunk 1]
```diff
@@ -0,0 +1,241 @@
+package identifierutil
```
Comment (warunalakshitha) [inline identifierutil/util.go:19]: Should we include licence header?
